packages/material-ui-system/src/createStyled.d.ts

```diff
@@ -142,44 +145,54 @@ export interface CreateStyledComponent<
 
 export interface CreateMUIStyled<Theme extends object = DefaultTheme> {
   <
-    C extends React.ComponentClass<React.ComponentProps<C>>,
+    ComponentProps extends {} = any,
+    Classes extends {} = any,
+    C extends React.ComponentClass<React.ComponentProps<C>> = any,
     ForwardedProps extends keyof React.ComponentProps<C> = keyof React.ComponentProps<C>,
   >(
     component: C,
-    options: FilteringStyledOptions<React.ComponentProps<C>, ForwardedProps> & MuiStyledOptions,
+    options: FilteringStyledOptions<React.ComponentProps<C>, ForwardedProps> &
+      MuiStyledOptions<ComponentProps, Classes>,
   ): CreateStyledComponent<
     Pick<PropsOf<C>, ForwardedProps> & {
       theme?: Theme;
       as?: React.ElementType;
       sx?: SxProps<Theme>;
     },
-    {},
+    ComponentProps,
     {
       ref?: React.Ref<InstanceType<C>>;
     }
   >;
 
-  <C extends React.ComponentClass<React.ComponentProps<C>>>(
+  <
+    ComponentProps extends {} = any,
+    Classes extends {} = any,
+    C extends React.ComponentClass<React.ComponentProps<C>> = any,
```

Comment:
Let’s add the new generics at the end. That way if they are not provides the signature is the same as emotion and styled components.